pmd-apex/src/test/java/net/sourceforge/pmd/lang/apex/metrics/ApexProjectMirrorTest.java

```diff
@@ -0,0 +1,157 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.apex.metrics;
+
+import static net.sourceforge.pmd.lang.apex.metrics.ApexMetricsVisitorTest.parseAndVisitForString;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+import org.junit.Test;
+
+import net.sourceforge.pmd.lang.apex.ast.ASTMethod;
+import net.sourceforge.pmd.lang.apex.ast.ASTUserClass;
+import net.sourceforge.pmd.lang.apex.ast.ASTUserClassOrInterface;
+import net.sourceforge.pmd.lang.apex.ast.ApexNode;
+import net.sourceforge.pmd.lang.apex.ast.ApexParserVisitorAdapter;
+import net.sourceforge.pmd.lang.apex.metrics.impl.AbstractApexClassMetric;
+import net.sourceforge.pmd.lang.apex.metrics.impl.AbstractApexOperationMetric;
+import net.sourceforge.pmd.lang.metrics.Metric.Version;
+import net.sourceforge.pmd.lang.metrics.MetricKey;
+import net.sourceforge.pmd.lang.metrics.MetricKeyUtil;
+import net.sourceforge.pmd.lang.metrics.MetricMemoizer;
+import net.sourceforge.pmd.lang.metrics.MetricVersion;
+
+import apex.jorje.semantic.ast.compilation.Compilation;
+
+/**
+ * @author Clément Fournier
+ */
+public class ApexProjectMirrorTest {
+
+    private static ApexNode<Compilation> acu
+        = parseAndVisitForString("public with sharing class MetadataDeployController \n"
```

Comment:
Same here - please put the source code in a separate file - oh, it seems to be actually the same file, so you could reuse the file "MetadataDeployController.cls".